tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.cs

```diff
@@ -96,6 +92,98 @@ internal readonly partial struct SecurityCoordinator
 
     internal static Span TryGetRoot(Span span) => span.Context.TraceContext?.RootSpan ?? span;
 
+    public IResult? RunWafForUser(string? userId = null, string? userLogin = null, string? userSessionId = null, bool fromSdk = false, Dictionary<string, string>? otherTags = null)
+    {
+        if (_httpTransport.IsBlocked)
+        {
+            return null;
+        }
+
+        IResult? result = null;
+        Dictionary<string, object>? addresses = null;
+        try
+        {
+            var additiveContext = GetOrCreateAdditiveContext();
+            if (additiveContext?.ShouldRunWith(_security, userId, userLogin, userSessionId, fromSdk) is { Count: > 0 } userAddresses)
+            {
+                addresses = userAddresses.ToDictionary(k => k.Key, object (v) => v.Value);
```

Comment:
`userAddresses` is _already_ a `Dictionary<>`, right, and isn't shared anywhere afaict? Can we avoid creating _another_ dictionary here, and just use the one provided instead?

Reply:
fixed in [fa8501f](https://github.com/DataDog/dd-trace-dotnet/pull/6431/commits/fa8501fcdd4f8f05559112b6bfe57669ed2fee09) thanks!